app/src/main/java/com/example/parkingproject/activities/MenuActivity.java

```diff
@@ -0,0 +1,69 @@
+package com.example.parkingproject.activities;
+
+import android.os.Bundle;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.FragmentTransaction;
+
+import com.example.parkingproject.R;
+import com.example.parkingproject.databinding.ActivityMenuBinding;
+import com.example.parkingproject.entities.Parking;
+import com.example.parkingproject.fragments.ReleaseFragment;
+import com.example.parkingproject.fragments.ReservationFragment.ReservationFragmentDelegate;
+import com.example.parkingproject.fragments.ReservationFragment;
+import com.example.parkingproject.mvp.presenter.MenuPresenter;
+
+import static com.example.parkingproject.fragments.ReservationFragment.PARKING_KEY;
+
+public class MenuActivity extends AppCompatActivity implements ReservationFragmentDelegate, ReleaseFragment.ReleaseFragmentDelegate {
+
+    private static final String RESERVATION_FRAGMENT_TAG = "RESERVATION_FRAGMENT";
+    private static final String RELEASE_FRAGMENT_TAG = "RELEASE_FRAGMENT";
+    public static final String PARKING_SIZE_KEY = "SIZE_KEY";
+
+    private ActivityMenuBinding binding;
+    private MenuPresenter menuPresenter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityMenuBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+        int parkingSize = getIntent().getIntExtra(PARKING_SIZE_KEY, 0);
+        menuPresenter = new MenuPresenter(parkingSize);
+        setListeners();
+    }
+
+    private void setListeners() {
+        ReservationFragment reservationFragment = new ReservationFragment();
+        ReleaseFragment releaseFragment = new ReleaseFragment();
```

Comment:
As these are expensive operations, it's better to do them just when needed, in the moment you are going to shown them (inside the click listeners).